### PR TITLE
[FLINK-2754] Add a new module for micro benchmark.

### PR DESCRIPTION
**Pull request by ChengXiangLi**
Add flink-benchmark module, and add JMH dependeny in pom.

### CODE REVIEW

**Comment by StephanEwen** [comment]
Looks good.

This makes only sense to merge, if we move the MemorySegment micro benchmarks and the XORShift benchmarks to this module. @ChengXiangLi Would you be up for that part?

Another question: Should the artifact of this project be deployed to maven central, or is this purely a project that people use when they work with the code?

**Comment by ChengXiangLi** [comment]
Yes, the XORShfit benchmark is ready to merge, and i would lead the work about other benchmark refactor.
No one should depends on this module, so i think there should be no requirement to deploy this module to maven central.

**Comment by uce** [comment]
Nice addition! :) I would also wait until someone asks for it.

File: flink-benchmark/pom.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      <scope>provided</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.2</version>
```
Comment (chiwanpark) [inline flink-benchmark/pom.xml]: Could you remove this version tag? The version of `maven-shade-plugin` is controlled by parent POM.

**Comment by chiwanpark** [comment]
Looks good to merge except one minor issue. Very nice start! :)

**Comment by StephanEwen** [comment]
Merging this...

**Pull request closed without merging**